lib/features/job/jobSlice.ts

```diff
@@ -58,8 +58,16 @@ export const selectJobCount = (state: RootState) => state.jobs.allIds.length
  * 
  * @param state 
  */
-export const selectTotalPaymentsByJob = (state: RootState, jobId: string) => 
-  selectPaymentsByJob(state, jobId).reduce((total: number, payment: PaymentItem) => total + payment.amount, 0)
+export const selectTotalPaymentsByJob = (state: RootState, jobId: string) => {
```

Comment:
i had to make this return both the data (as an object `byId`) and the iterator (`allIds`)